src/game/UnitAuraProcHandler.cpp

```diff
@@ -2347,9 +2347,9 @@ SpellAuraProcResult Unit::HandleMendingAuraProc(Unit* /*pVictim*/, uint32 /*dama
                 triggeredByAura->SetInUse(true);
                 RemoveAurasByCasterSpell(spellProto->Id, caster->GetObjectGuid());
 
-                caster->AddSpellMod(mod, true);
+                ((Player*)this)->AddSpellMod(mod, true);
                 CastCustomSpell(target, spellProto->Id, &heal, nullptr, nullptr, TRIGGERED_OLD_TRIGGERED, nullptr, triggeredByAura, caster->GetObjectGuid());
-                caster->AddSpellMod(mod, false);
+                ((Player*)this)->AddSpellMod(mod, false);
```

Comment:
Yup, we were checking wrong target for the stack reduction spellmod. Its on the prev target, not on player.